pkg/auth/multiauthenticator.go

```diff
@@ -0,0 +1,67 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"strings"
+)
+
+type IssuerWithClientID struct {
+	Issuer   string // Issuer of the token.
+	ClientID string // ClientID of the token, can be empty.
+}
+
+type multiAuthenticatorWithClientIDImpl struct {
+	authenticators map[IssuerWithClientID]Authenticator
+}
+
+func NewMultiAuthenticatorWithClientID(
+	authenticators map[IssuerWithClientID]Authenticator) MultiAuthenticatorWithClientID {
+	return &multiAuthenticatorWithClientIDImpl{
+		authenticators: authenticators,
+	}
+}
+
+// GetAuthenticator returns nil if there is no authenticator for the given issuer and client ID.
+func (m *multiAuthenticatorWithClientIDImpl) GetAuthenticator(issuer, audience string) Authenticator {
+	for issuerWithClientID, authenticator := range m.authenticators {
+		// Audience field in a claim can contain more than just the clientID.
+		// The strings.Contains check here is same as what is being used by the OIDC library
+		// which verifies the token.
+		if issuer == issuerWithClientID.Issuer && strings.Contains(audience, issuerWithClientID.ClientID) {
```

Comment:
When using Contains(), should we worry about overlapping clientID strings? e.g. `abc`, `abcd`